### ISSUE TITLE
--daemon & --audit together multiplies responses and related issues

### ISSUE THREAD

**Issue by arielf** [JohnLangford/vowpal_wabbit#1397]
When I try to add `--audit` option to `--daemon` the audit printout is printed multiple times.

In addition:
- The audit duplication factor is 13 (larger than the number of vw children subprocesses which is 10)
- The audit output is not the same in all 13 repetitions, even though I used `-t` (read-only model)
- At the end, the progress line gets mixed with the tag/prediction output in its middle (see below)

```
$ vw --daemon -t -a  -i somefile.model
Generating 2-grams for all namespaces.
Generating 1-skips for all namespaces.
only testing
Num weight bits = 24
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
$ calling accept

$ echo "1 'example1|ABC | z" | netcat localhost 26542
reading data from port 26542
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.035107 example1
        Constant:1856960:1:0.0351067    z:6922608:1:0
0.032295 example1
        Constant:1856961:1:0.0322954    z:6922609:1:0
0.033821 example1
        Constant:1856962:1:0.0338212    z:6922610:1:0
0.032240 example1
        Constant:1856963:1:0.0322399    z:6922611:1:0
0.034082 example1
        Constant:1856964:1:0.0340821    z:6922612:1:0
0.034929 example1
        Constant:1856965:1:0.0349287    z:6922613:1:0
0.033841 example1
        Constant:1856966:1:0.0338413    z:6922614:1:0
0.034339 example1
        Constant:1856967:1:0.034339     z:6922615:1:0
0.033310 example1
        Constant:1856968:1:0.0333102    z:6922616:1:0
0.035687 example1
        Constant:1856969:1:0.0356869    z:6922617:1:0
0.034944 example1
        Constant:1856970:1:0.0349442    z:6922618:1:0
0.035495 example1
        Constant:1856971:1:0.0354953    z:6922619:1:0
0.034815 example1
        Constant:1856972:1:0.0348148    z:6922620:1:0
0.932724 0.932724            10.034224 example1 0.032240 0.035687
            1.0   1.0000   0.0342        2
```

**Comment by JohnLangford** [comment]
I can't replicate this.  In particular, I tried:
$ echo "1 'example1|ABC | z" | vw -f foo
final_regressor = foo
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = 
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
1.000000 1.000000            1            1.0   1.0000   0.0000        1

finished run
number of examples per pass = 1
passes used = 1
weighted example sum = 1.000000
weighted label sum = 1.000000
average loss = 1.000000
best constant = 1.000000
best constant's loss = 0.000000
total feature number = 1
$ ./vw --daemon -t -a -i fooonly testing
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
calling accept
$ echo "1 'example1|ABC | z" | netcat localhost 26542
0.393395 example1
	z:170519:1:0.393395	Constant:116060:1:0
0.393395 example1

This seems to be the expected output---the first two lines are audit from the daemon while the last line is the return from netcat.
